src/main/resources/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTrigger/config.jelly

```diff
@@ -6,6 +6,32 @@
   <f:entry title="Stash URL" field="stashHost">
       <f:textbox />
   </f:entry>
+  <f:nested>
+    <table>
+      <f:optionalBlock title="Enable proxy" name="proxySettings" checked="${instance.getProxySettings() != null}">
+        <f:nested>
+          <table>
+            <f:radioBlock name="proxyChoice" value="jenkins" title="Inherit from Jenkins" checked="${(instance.getProxySettings() != null) ? instance.isProxyType('jenkins') : true}">
+            </f:radioBlock>
+            <f:radioBlock name="proxyChoice" value="custom" title="Custom" checked="${(instance.getProxySettings() != null) ? instance.isProxyType('custom') : false}">
+              <f:entry title="Proxy Host" field="proxyHost">
+                <f:textbox value="${(instance.getProxySettings() != null &amp;&amp; instance.isProxyType('custom')) ? instance.getProxyDetails().getString('proxyHost') : ''}" />
```

Comment:
there is significant repetition here
I would expect jelly has some kind of a block/panel element that would enable you to hide irrelevant fields with one  toggle